### PR TITLE
Added configuration for UAT environment

### PR DESCRIPTION
**Pull request by stuartmccoll**
## Proposed changes

This commit adds the necessary configuration for the newly-added UAT environment in GOV.UK PaaS.

Initial configuration has been copied from the existing staging configuration.

## Further comments

Note that this doesn't yet include updates to the GitHub Action. I'm happy to manually push for the moment (given the small amount of changes we're likely to be making in the next sprint), whilst we discuss potential changes to the pipeline workflow.

## Screenshot (if applicable)

N/A

## Checklist

- [ ] Added tests for new functionality, or updated existing tests
- [ ] Updated `README.md` if necessary

### CODE REVIEW

File: config/environments/uat.rb [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  
+    # Enable serving of images, stylesheets, and JavaScripts from an asset server.
+    # config.action_controller.asset_host = 'http://assets.example.com'
+  
+    # Specifies the header that your server uses for sending files.
+    # config.action_dispatch.x_sendfile_header = 'X-Sendfile' # for Apache
+    # config.action_dispatch.x_sendfile_header = 'X-Accel-Redirect' # for NGINX
+  
+    # Store uploaded files on the local file system (see config/storage.yml for options).
+    config.active_storage.service = :local
```
Comment (gidsg) [inline config/environments/uat.rb:39]: I think this line can be removed (and also from the staging and production configs) as it gets overriden further down.

File: config/environments/uat.rb [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    }
+  
+    config.x.ideal_postcodes.api_key = ENV.fetch("IDEAL_POSTCODES_API_KEY")
+    config.x.salesforce.username = ENV.fetch("SALESFORCE_USERNAME")
+    config.x.salesforce.password = ENV.fetch("SALESFORCE_PASSWORD")
+    config.x.salesforce.security_token = ENV.fetch("SALESFORCE_SECURITY_TOKEN")
+    config.x.salesforce.client_id = ENV.fetch("SALESFORCE_CLIENT_ID")
+    config.x.salesforce.client_secret = ENV.fetch("SALESFORCE_CLIENT_SECRET")
+    # TODO: update this with production host once credentials available
+    config.x.salesforce.host = "test.salesforce.com"
```
Comment (gidsg) [inline config/environments/uat.rb:138]: I think we need to change this to the UAT host? Probably should be an env var in this instance.

Comment (stuartmccoll) [reply config/environments/uat.rb:138]: I thought the development and UAT Salesforce environments lived under the same host? We tested our UAT environment manually and it ended up in UAT Salesforce.

Comment (gidsg) [reply config/environments/uat.rb:138]: OK I wasn't sure about this but fine if you have tested it.

**Comment by gidsg** [comment]
@stuartmccoll whilst we're looking at this could we also remove this from the other environments e.g. https://github.com/heritagefund/funding-frontend/blob/75c865bb9e3bb4d0b4829c2be155ef5c028318ab/config/environments/staging.rb#L39
and
https://github.com/heritagefund/funding-frontend/blob/75c865bb9e3bb4d0b4829c2be155ef5c028318ab/config/environments/production.rb#L42
?

**Pull request merged**